autofs-5.1.1 - fix use after free in open_lookup()

From: Ian Kent <raven@themaw.net>

If storage can't be allocated for module type error exit.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG       |    1 +
 daemon/module.c |    1 +
 2 files changed, 2 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index 76e0a27..2d026f1 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -36,6 +36,7 @@
 - fix memory leak in nisplus lookup_reinit().
 - fix memory leak in ldap do_init().
 - fix use after free in sun parser parse_init().
+- fix use after free in open_lookup().
 
 21/04/2015 autofs-5.1.1
 =======================
diff --git a/daemon/module.c b/daemon/module.c
index d9921f4..bed8f7a 100644
--- a/daemon/module.c
+++ b/daemon/module.c
@@ -83,6 +83,7 @@ int open_lookup(const char *name, const char *err_prefix, const char *mapfmt,
 			char *estr = strerror_r(errno, buf, MAX_ERR_BUF);
 			logerr("%s%s", err_prefix, estr);
 		}
+		return NSS_STATUS_UNAVAIL;
 	}
 
 	size = snprintf(fnbuf, sizeof(fnbuf),
